File: Godot-MCP/server/src/llm-connectors/openai_client.ts
import OpenAI from 'openai';
import { BaseLlmClient } from './base_client';

export class OpenAIClient implements BaseLlmClient {
    private readonly client: OpenAI;
    private readonly model: string;

    constructor() {
        const apiKey = process.env.OPENAI_API_KEY;
        this.model = process.env.OPENAI_MODEL || 'gpt-4-turbo';

        if (!apiKey || apiKey.includes('YOUR_OPENAI_API_KEY_HERE')) {
            throw new Error('OpenAI API key is missing or not set in .env file');
        }
        this.client = new OpenAI({ apiKey });
    }

    async generatePlan(systemPrompt: string, userPrompt: string): Promise<string> {
        const response = await this.client.chat.completions.create({
            model: this.model,
            messages: [
                { role: "system", content: systemPrompt },
                { role: "user", content: userPrompt },
            ],
            response_format: { type: "json_object" },
        });
        return response.choices[0].message.content || '';
    }
}

File: bmad_method/llm_connectors/openai_client.py
import os
from openai import OpenAI
from .base_client import BaseLlmClient

class OpenAIClient(BaseLlmClient):
    def __init__(self):
        super().__init__()
        self.api_key = os.getenv('OPENAI_API_KEY')
        self.model = os.getenv('OPENAI_MODEL')
        if not self.api_key or 'YOUR_OPENAI_API_KEY_HERE' in self.api_key:
            raise ValueError("OpenAI API key is missing or not set in .env file")
        self.client = OpenAI(api_key=self.api_key)

    def generate_plan(self, system_prompt: str, user_prompt: str) -> str:
        response = self.client.chat.completions.create(
            model=self.model,
            messages=[
                {"role": "system", "content": system_prompt},
                {"role": "user", "content": user_prompt}
            ],
            response_format={"type": "json_object"}
        )
        return response.choices[0].message.content
